modAionImpl/src/org/aion/zero/impl/blockchain/AionImpl.java

```diff
@@ -67,6 +67,7 @@ private AionImpl(boolean forTest) {
 
         blockchainCallbackInterfaces = Collections.synchronizedList(new ArrayList<>());
         aionHub.getPendingState().setPendingTxCallback(new PendingTxCallback(blockchainCallbackInterfaces));
+        aionHub.getPendingState().setNetworkBestBlockNumberCallback(new NetworkBestBlockCallback(this));
```

Comment:
I think this should be moved to the pending state constructor. This would remove the need for the setter and the NPE worries.